sysl2/sysl/seqs/writer_test.go

```diff
@@ -0,0 +1,302 @@
+
+	// when
+	w.WriteHead("head")
+
+	// then
+	assert.Equal(t, 5, w.head.Len())
+}
+
+func TestIndent(t *testing.T) {
+	// given
+	w := MakeSequenceDiagramWriter(true)
+
+	// when
+	w.Indent()
+
+	// then
+	assert.Equal(t, 1, w.ind)
+}
+
+func TestUnindent(t *testing.T) {
+	w := MakeSequenceDiagramWriter(true)
+
+	w.Unindent()
+	assert.Equal(t, 0, w.ind)
+
+	w.Indent()
+	assert.Equal(t, 1, w.ind)
+
+	w.Unindent()
+	assert.Equal(t, 0, w.ind)
+
+	w.Unindent()
+	assert.Equal(t, 0, w.ind)
+}
+
+func TestActivate(t *testing.T) {
+	w := MakeSequenceDiagramWriter(true)
+
+	w.Activate("a")
+
+	assert.Equal(t, 11, w.body.Len())
+	assert.True(t, w.atBeginOfLine)
+	assert.Equal(t, map[string]int{"a": 1}, w.active)
+
+	w.Activate("a")
+	assert.Equal(t, 22, w.body.Len())
+	assert.True(t, w.atBeginOfLine)
+	assert.Equal(t, map[string]int{"a": 2}, w.active)
+}
+
+func TestActivated(t *testing.T) {
+	w := MakeSequenceDiagramWriter(true)
+
+	d := w.Activated("a", false)
+
+	assert.Equal(t, 11, w.body.Len())
+	assert.True(t, w.atBeginOfLine)
+	assert.Equal(t, map[string]int{"a": 1}, w.active)
+
+	d()
+	assert.Equal(t, 24, w.body.Len())
+	assert.True(t, w.atBeginOfLine)
+	assert.Equal(t, map[string]int{}, w.active)
+
+	d()
+	assert.Equal(t, 24, w.body.Len())
+	assert.True(t, w.atBeginOfLine)
+	assert.Equal(t, map[string]int{}, w.active)
+}
+
+func TestActivatedWithSuppressed(t *testing.T) {
+	w := MakeSequenceDiagramWriter(true)
+
+	d := w.Activated("a", true)
+
+	assert.Equal(t, 0, w.body.Len())
+	assert.True(t, w.atBeginOfLine)
+	assert.Equal(t, map[string]int{}, w.active)
+
+	d()
+	assert.Equal(t, 0, w.body.Len())
+	assert.True(t, w.atBeginOfLine)
+	assert.Equal(t, map[string]int{}, w.active)
+
+	d()
+	assert.Equal(t, 0, w.body.Len())
+	assert.True(t, w.atBeginOfLine)
+	assert.Equal(t, map[string]int{}, w.active)
+}
+
+func TestDeactivate(t *testing.T) {
+	w := MakeSequenceDiagramWriter(true)
+
+	w.Activate("a")
+
+	assert.Equal(t, 11, w.body.Len())
+	assert.True(t, w.atBeginOfLine)
+	assert.Equal(t, map[string]int{"a": 1}, w.active)
+
+	w.Deactivate("a")
+	assert.Equal(t, 24, w.body.Len())
+	assert.True(t, w.atBeginOfLine)
+	assert.Equal(t, map[string]int{}, w.active)
+
+	w.Deactivate("a")
+	assert.Equal(t, 24, w.body.Len())
+	assert.True(t, w.atBeginOfLine)
+	assert.Equal(t, map[string]int{}, w.active)
+
+	w.Deactivate("b")
+	assert.Equal(t, 24, w.body.Len())
+	assert.True(t, w.atBeginOfLine)
+	assert.Equal(t, map[string]int{}, w.active)
+}
+
+func TestWriteIndent(t *testing.T) {
+	w := MakeSequenceDiagramWriter(true)
+	w.atBeginOfLine = true
+	w.ind = 1
+
+	w.writeIndent()
+
+	assert.Equal(t, 1, w.body.Len())
+	assert.False(t, w.atBeginOfLine)
+
+	w.writeIndent()
+	assert.Equal(t, 1, w.body.Len())
+	assert.False(t, w.atBeginOfLine)
+}
+
+func TestStringer(t *testing.T) {
+	w := MakeSequenceDiagramWriter(false)
+	w.WriteHead("head")
+	_, err := w.WriteString("body\n")
+
+	s := w.String()
+	expected := "@startuml\nhead\nbody\n@enduml\n"
```

Comment:
Needless local vars. Use the expressions inline in `assert.Equal(…)`. Ditto below (for short expressions only).